_tools/types/main.go

```diff
@@ -0,0 +1,333 @@
+// Generates a report about every driver wich UASTv2 types does it use,
+// both in actual test fixtures and though the source code.
+package main
+
+import (
+	"bufio"
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	_ "net/http/pprof"
+	"os"
+	"os/exec"
+	"path"
+	"reflect"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/bblfsh/sdk/driver/manifest/discovery"
+	"github.com/bblfsh/sdk/uast"
+)
+
+const (
+	fixtureDir = "fixtures"
+	fixtureExt = ".sem.uast"
+	goDocUrl   = "https://godoc.org/github.com/bblfsh/sdk/uast#"
+)
+
+var (
+	reposRootPath = path.Join(".", "drivers")
+	pprof         = flag.Bool("pprof", false, "start pprof profiler http endpoing")
+	skipUpdate    = flag.Bool("skip", false, "skip git clone or pull")
+)
+
+func main() {
+	flag.Parse()
+
+	if *pprof {
+		pprofAddr := "localhost:6060"
+		fmt.Fprintf(os.Stderr, "running pprof on %s\n", pprofAddr)
+		go func() {
+			if err := http.ListenAndServe(pprofAddr, nil); err != nil {
+				log.Fatal("cannot start pprof: %v", err)
+			}
+		}()
+	}
+
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run() error {
+	drivers, err := listDrivers()
+	if err != nil {
+		return fmt.Errorf("failed to list drivers: %s", err)
+	}
+
+	if !*skipUpdate {
+		err = maybeCloneOrPullAll(drivers)
+		if err != nil {
+			return fmt.Errorf("failed to pull driver repos: %s", err)
+		}
+	}
+
+	uastTypes := findUastTypesInSdk()
+	for _, driver := range drivers {
+		if err := analyzeFixtures(driver); err != nil {
+			return err
+		}
+		analyzeCode(driver, uastTypes)
+	}
+
+	formatMarkdownTable(drivers, uastTypes)
+	return nil
+}
+
+type driverStats struct {
+	url          string
+	language     string
+	path         string
+	skip         bool
+	fixturesNum  int
+	fixturesUast map[string]int
+	codeUast     map[string]int
+}
+
+// listDrivers lists all available drivers.
+func listDrivers() ([]*driverStats, error) {
+	fmt.Fprintf(os.Stderr, "discovering all available drivers\n")
+	langs, err := discovery.OfficialDrivers(context.TODO(), &discovery.Options{
+		NoStatic: true,
+	})
+	if err != nil {
+		return nil, err
+	}
+	drivers := make([]*driverStats, 0, len(langs))
+	for _, l := range langs {
+		drivers = append(drivers, &driverStats{
+			language:     l.Language,
+			url:          l.RepositoryURL(),
+			path:         l.RepositoryURL()[strings.LastIndex(l.RepositoryURL(), "/"):],
+			fixturesUast: make(map[string]int),
+			codeUast:     make(map[string]int),
+		})
+	}
+	fmt.Fprintf(os.Stderr, "%d drivers available on-line\n", len(langs))
+	return drivers, nil
+}
+
+// maybeCloneOrPullAll either clones repos to path in local FS or, if already preset,
+// pulls the latest master for each of them.
+func maybeCloneOrPullAll(drivers []*driverStats) error {
+	fmt.Fprintf(os.Stderr, "cloning %d drivers to %s\n", len(drivers), reposRootPath)
+	err := os.MkdirAll(reposRootPath, os.ModePerm)
+	if err != nil {
+		return err
+	}
+
+	var (
+		cr = make(chan int, 3)
```

Comment:
```suggestion
		cr = make(chan int, 3) // fetches to run in parallel
```
Or maybe there is a better name that can be used here.